aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -241,52 +272,83 @@ void _validate_compressed_sparse_indices_kernel(
 
     const auto batch_dims = cidx.sizes().slice(0, cidx.dim() - 1);
     const auto batch_count = indexCount(batch_dims);
-    const auto batch_idx = at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+    const auto batch_idx =
+        at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+
+    const auto idx_ndims = idx.dim();
+    const auto idx_sizes = idx.sizes();
+    const auto idx_strides = idx.strides();
+    const auto idx_contiguous_strides = c10::contiguous_strides(idx_sizes);
+    std::array<int64_t, 3 * MAX_DIMS> idx_sizes_and_strides;
```

Comment:
nit: having 3 separate arrays might be a bit easier to understand.

Reply:
Fixed.